src/service/data_products/common_models.py

```diff
@@ -77,11 +82,9 @@ def _check_router_tags(cls, v):  # @NoSelf
     @field_validator("db_collections")
     @classmethod
     def _ensure_only_one_view(cls, v):
-        found = False
-        for dbc in v:
-            if found and v.view_required:
-                raise ValueError("More than one db collection requiring a view found")
-            found = found or dbc.view_required
+        view_required_collections = [dbc for dbc in v if dbc.view_required or dbc.generic_view_required]
+        if len(view_required_collections) > 1:
+            raise ValueError("More than one db collection requiring a view found")
```

Comment:
So does this actually work? now I distrust how the method is set up since the prior version had a bug in it that seems like it should have puked if the method was run

Reply:
I think it's working as expected. 

`
In [10]: DataProductSpec(
    ...:         data_product="test_product",
    ...:         db_collections=[
    ...:             DBCollection(name="Collection1", view_required=True, generic_view_require
    ...: d=False, indexes=[]),
    ...:             DBCollection(name="Collection2", view_required=False, generic_view_requir
    ...: ed=True, indexes=[]),
    ...:         ],
    ...:         router=APIRouter(tags=["tag1"]),
    ...:     )
---------------------------------------------------------------------------
ValidationError                           Traceback (most recent call last)
Cell In[10], line 1
----> 1 DataProductSpec(
      2         data_product="test_product", 
      3         db_collections=[
      4             DBCollection(name="Collection1", view_required=True, generic_view_required=False, indexes=[]),
      5             DBCollection(name="Collection2", view_required=False, generic_view_required=True, indexes=[]),
      6         ],
      7         router=APIRouter(tags=["tag1"]),
      8     )

File /opt/miniconda3/envs/dev/lib/python3.11/site-packages/pydantic/main.py:164, in BaseModel.__init__(__pydantic_self__, **data)
    162 # `__tracebackhide__` tells pytest and some other tools to omit this function from tracebacks
    163 __tracebackhide__ = True
--> 164 __pydantic_self__.__pydantic_validator__.validate_python(data, self_instance=__pydantic_self__)

ValidationError: 1 validation error for DataProductSpec
db_collections
  Value error, More than one db collection requiring a view found [type=value_error, input_value=[DBCollection(name='Colle...uired=True, indexes=[])], input_type=list]
    For further information visit https://errors.pydantic.dev/2.5/v/value_error
`

Reply:
If I import the code exactly as it is in the main branch, it will raise an expected error: 'list' object has no attribute 'view_required'. Surprisingly, this issue was never not caught.

![Screenshot 2023-12-15 at 4 13 41 PM](https://github.com/kbase/collections/assets/3953823/0f53fab7-9d5e-45fe-b527-9323b93a653b)

Reply:
yeah, I don't understand how `genome_attribs` was ever able to import like that. It should've choked

Reply:
Oh, I bet I know. The `and` causes the `if` to short circuit and since the `dbc` with `view_required` is last there's never a loop with `found==True`

Reply:
That means I either never tested it or screwed up the test though...

Reply:
yea. That's exactly what happened. `found` was never True. If we put 'genome_attribs' before 'genome_attribs_meta', this bug will appear.